### PR TITLE
PR: Fix error when trying to add a non-ascii module in Python 2 to the UMR blacklist

### LINKED ISSUE #3553: Spyder fails to launch because of Unicode errors in create_kernel_spec
## Spyder Launch Error

When I run spyder on command line, I got error message like below.

> Traceback (most recent call last):   File
> "/home/park/anaconda2/lib/python2.7/site-packages/spyder/app/mainwindow.py",
> line 2998, in main
>     mainwindow = run_spyder(app, options, args)   File "/home/park/anaconda2/lib/python2.7/site-packages/spyder/app/mainwindow.py",
> line 2902, in run_spyder
>     main.setup()   File "/home/park/anaconda2/lib/python2.7/site-packages/spyder/app/mainwindow.py",
> line 1153, in setup
>     self.setup_layout(default=False)   File "/home/park/anaconda2/lib/python2.7/site-packages/spyder/app/mainwindow.py",
> line 1414, in setup_layout
>     self.setup_default_layouts('default', settings)   File "/home/park/anaconda2/lib/python2.7/site-packages/spyder/app/mainwindow.py",
> line 1593, in setup_default_layouts
>     widget.toggle_view(True)   File "/home/park/anaconda2/lib/python2.7/site-packages/spyder/plugins/ipythonconsole.py",
> line 677, in toggle_view
>     self.create_new_client(give_focus=False)   File "/home/park/anaconda2/lib/python2.7/site-packages/spyder/plugins/ipythonconsole.py",
> line 886, in create_new_client
>     self.connect_client_to_kernel(client)   File "/home/park/anaconda2/lib/python2.7/site-packages/spyder/plugins/ipythonconsole.py",
> line 903, in connect_client_to_kernel
>     km, kc = self.create_kernel_manager_and_kernel_client(connection_file)   File
> "/home/park/anaconda2/lib/python2.7/site-packages/spyder/plugins/ipythonconsole.py",
> line 1276, in create_kernel_manager_and_kernel_client
>     kernel_manager._kernel_spec = self.create_kernel_spec()   File "/home/park/anaconda2/lib/python2.7/site-packages/spyder/plugins/ipythonconsole.py",
> line 1256, in create_kernel_spec
>     uv = to_text_string(v)   File "/home/park/anaconda2/lib/python2.7/site-packages/spyder/py3compat.py",
> line 136, in to_text_string
>     return unicode(obj) UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 73: ordinal not in range(128)

**I have tried**
1. Reinstall using "pip install spyder"
2. Reinstall using "conda install spyder"
3. On the command line "spyder --reset" 

**Version Information**(
- Spyder Version: 3.0
- Python Version: 2.7.12
- Operating system: Ubuntu 14.04
- Anaconda Version : 4.2.0

How do I fix it. If you give me some ideas, I can really appreciate for this.

Thanks.

### PR DESCRIPTION
**Pull request by mariacamilaremolinagutierrez**
Fixes #3553

### CODE REVIEW

**Comment by mariacamilaremolinagutierrez** [comment]
It is not possible to import a non-ascii module in Python 2. For this reason now there s warning message that doesn't allow the user to import it because it will cause an error. 

a

File: spyder/plugins/ipythonconsole.py [hunk 1]
```diff
@@ -1238,13 +1238,22 @@ def create_kernel_spec(self):
         ]
 
         # Environment variables that we need to pass to our sitecustomize
+        umr_namelist_processed = CONF.get('main_interpreter', 'umr/namelist')
```
Comment (ccordoba12) [inline spyder/plugins/ipythonconsole.py]: Let's simplify this variable name :-) Please call it just `umr_namelist`

File: spyder/plugins/ipythonconsole.py [hunk 2]
```diff
@@ -1238,13 +1238,22 @@ def create_kernel_spec(self):
...
         # Environment variables that we need to pass to our sitecustomize
+        umr_namelist_processed = CONF.get('main_interpreter', 'umr/namelist')
+
+        if PY2:
+            for umr_n_p in umr_namelist_processed:
+                try:
+                    umr_n_p.encode('utf-8')
+                except UnicodeDecodeError:
+                    umr_namelist_processed.remove(umr_n_p)
+            CONF.set('main_interpreter', 'umr/namelist', umr_namelist_processed)
```
Comment (ccordoba12) [inline spyder/plugins/ipythonconsole.py]: Nice!!

File: spyder/plugins/maininterpreter.py [hunk 3]
```diff
@@ -184,9 +184,23 @@ def set_umr_namelist(self):
...
+                        else:
+                            QMessageBox.warning(self, _('Warning'),
+                            _("You are working with Python 2, this means that "
+                              "you can not import a module that contains non-"
+                              "ascii characters."), QMessageBox.Ok)
+                            non_ascii_namelist.append(module_name)
+                    elif programs.is_module_installed(module_name):
+                        fixed_namelist.append(module_name)
+                invalid = ", ".join(set(namelist)-set(fixed_namelist)-
+                                    set(non_ascii_namelist))
```
Comment (ccordoba12) [inline spyder/plugins/maininterpreter.py]: I really like this, good job!

**Comment by ccordoba12** [comment]
@mariacamilaremolinagutierrez, just one minor comment and this ready to be merged :-)

File: spyder/plugins/ipythonconsole.py (hunk 1 above)
Comment (mariacamilaremolinagutierrez) [inline spyder/plugins/ipythonconsole.py]: Done :)

**Comment by mariacamilaremolinagutierrez** [comment]
@ccordoba12 Thank you for the comments, it is ready now :)

File: spyder/plugins/ipythonconsole.py [hunk 4]
```diff
@@ -1238,13 +1238,22 @@ def create_kernel_spec(self):
...
         # Environment variables that we need to pass to our sitecustomize
+        umr_namelist = CONF.get('main_interpreter', 'umr/namelist')
+
+        if PY2:
+            for umr_n in umr_namelist:
+                try:
+                    umr_n.encode('utf-8')
+                except UnicodeDecodeError:
+                    umr_namelist.remove(umr_n)
+            CONF.set('main_interpreter', 'umr/namelist', umr_namelist)
```
Comment (ccordoba12) [inline spyder/plugins/ipythonconsole.py]: I think we should be a bit smarter here and do this only if the result after removing the unicode modules is different from the original one.

So @mariacamilaremolinagutierrez, please make a copy of `umr_namelist` inside PY2 but before the `for` loop and apply CONF.set only if `umr_namelist != original_list`

Comment (mariacamilaremolinagutierrez) [inline spyder/plugins/ipythonconsole.py]: Hi @ccordoba12 I am sorry I didn't see this on time, I just committed this change

**Comment by ccordoba12** [comment]
Thanks @mariacamilaremolinagutierrez, great job!

**Pull request merged**